### ISSUE TITLE
Module that deals cards

### ISSUE THREAD

**Issue by AmadeusW** [AmadeusW/Poker#2]

**Comment by Morkaleb** [comment]
We have one small issue that we need to resolve for our game.  The primary one really.  We talked about it briefly on Wednesday, but I don't think that we resolved it.  What kind of poker are we going to play.  I can work out a lot of the code with out us making a decision on this topic, but I can't finish anything without it.  I am thinking that we should go with Texas Holdem, simply because that seems t be the most popular style and in the "unlikely" event that we decide to publish our project, that would be the best way to go.  I can start building it that way and change it later.  Just let me know what everyone's feelings are on the subject.

**Comment by Morkaleb** [comment]
Also, we could always add other styles in the future, which might be fun as we build a more capable Ai into it:)

**Comment by AmadeusW** [comment]
:+1: for Texas Holdem.
We will definitely publish it when we finish it and it works :)

**Comment by AmadeusW** [comment]
we will need module that analyzes cards on a web-poker table via screenshots, and plays. If it makes money, we'll go out and drink a lot of beer :)

**Comment by Morkaleb** [comment]
"if it works" lol, of course it will work, the worst case scenario is that we will have to take a little bit of time to make it work.  That's not such a big concern, making it work.  I am stubborn :)

**Comment by Morkaleb** [comment]
So I have a portion of this done.  I have built a setup which ensures that each card is only issued once.  It also occurred to me that poker can not be played with more than one deck at a time, as there is no hand with 5 of a kind.  Inside the dealing class I have also included some value assignments to the cards.  Tomorrow I plan on building a hand value comparative algorithm.  I will also get started on building a basic AI which will compare basic behavior, like raising bets, to probability of having a a better hand than the dealer's.   I  think that when we have the larger coding framework setup, much of this code can be pasted in to this better organization.

**Comment by Morkaleb** [comment]
One thing that I don't know how to do that I will need to look up , is a way to save a game.   This will be important when we start to allow the player's behaviour to build the learning set for the AI.

**Comment by Morkaleb** [comment]
Ok, I have put my work up on GitHub (I hope that I've done it right, I'm still learning my way around.  At least that this point, I can't ruin anything else that was on there, since there wasn't anything on there yet :D What I have there is a VERY basic "DUI", Amadeus, I recall that you said that there is a way to only update the parts on the screenUpdate(int) that change, I don't know how to do that inside of a console though, It wouldn't be hard to build it inside a window, but I don't know how to do that either.  I was thinking that is for version 0.2 anyways.  The suitInt and CardValue are numbers that I plan to use to develop the hand calculations, which is something that I hope to work out how to build tomorrow.  I have some psuedo code in my head and I think that I can translate that into c# code with sufficient cups of coffee and head on desk banging :D.    SO far, it doesn't do much, but it does run.  I hope to finish the dealing/Raising/Calling part of it for the Human player tonight.  If I get that part done I'll post it before I go to bed.  Otherwise I'll be posting some of it tomorrow.

**Comment by AmadeusW** [comment]
For saving games we can use the `[Serializable]` attribute which allows the [serialization](https://msdn.microsoft.com/en-us/library/ms973893.aspx) code to store all object's properties and load them at a different time

Don't worry about the way the console draws. We're not concerned in frames per second or anything. As long as it shows the cards, it's a good GUI :)

**Comment by Morkaleb** [comment]
Yeah, for practical reasons for this project it isn't important,  but for learning reasons it's something that I would like to look into.  I'll read up on serialization this evening

**Comment by Morkaleb** [comment]
Yeah, the saving games will be really important when we start to build the
AI, since it will take more than 5 hands for the dealer to figure out how
to predict the way that you play.
